firedrake/mesh.py

```diff
@@ -106,6 +106,15 @@ def subset(self, markers):
         try:
             return self._subsets[markers]
         except KeyError:
+            # check that the given markers are valid
+            for marker in markers:
+                if marker not in self.markers:
```

Comment:
I think the parallel issue goes away if this tests in `self.unique_markers` rather than `self.markers`.